src/coreclr/tools/Common/Compiler/NativeAotNameMangler.cs

```diff
@@ -121,14 +117,11 @@ private string SanitizeNameWithHash(string literal)
                 byte[] hash;
                 lock (this)
                 {
-                    if (_sha256 == null)
-                    {
-                        // Use SHA256 hash here to provide a high degree of uniqueness to symbol names without requiring them to be long
-                        // This hash function provides an exceedingly high likelihood that no two strings will be given equal symbol names
-                        // This is not considered used for security purpose; however collisions would be highly unfortunate as they will cause compilation
-                        // failure.
-                        _sha256 = SHA256.Create();
-                    }
+                    // Use SHA256 hash here to provide a high degree of uniqueness to symbol names without requiring them to be long
+                    // This hash function provides an exceedingly high likelihood that no two strings will be given equal symbol names
+                    // This is not considered used for security purpose; however collisions would be highly unfortunate as they will cause compilation
+                    // failure.
+                    _sha256 ??= SHA256.Create();
```

Comment:
You can avoid the instance entirely and switch to using the one-shot static methods.